### ISSUE TITLE
Binding listener to IPv6 instead of IPv4

### ISSUE THREAD

**Issue by SechyMonkey** [Terence-D/GameInputCommandServer#32]
Running both IPv4 and IPv6, GIC Server binds the listener to the default IPv6 address instead of the IPv4 address and thus doesn't allow connection from the android application. I attempted inputting the IPv6 address on the android side without success. It was accepted, but no connection occurred. 

This might be corrected by allowing selection/input of the IP address to bind to on the server side, or by allowing selection of which IP version to utilize for the listener address?

**Comment by Terence-D** [comment]
Yeah, I think an ability to select would be best, I'll prioritize this and hope to have something by end of the weekend.  Thanks!

**Comment by Terence-D** [comment]
Hmm I checked the code, it should be binding to all adapters, see here:

```public bool Start()
        {...
            string baseAddress = "http://" + "*" + ":" + Port + "/";
...
        }
```

could it be a firewall setting?  Try (temporarily!) disabling any firewall you have on the server to see if a connection can get through.

**Comment by SechyMonkey** [comment]
Interesting. I gave it a try after disabling firewall, but no different behavior. I then removed it and extracted it from the zip file fresh and found that using the default port it was able to bind to default IPv4 and IPv6 correctly. Then I stopped and closed, re-opened and used a non-standard port (DOOM game server port, 666) and I saw no listeners on that port (though I did see one on 8093?). I then stopped and closed it again, and started with a different port (12366, which is what I was using previously) and it worked as expected this time. Same with firewall re-enabled, so that apparently wasn't the issue.

So possibly there's an intermittent issue with binding to non-default ports? Thank you for taking a look at this so quickly, by the way. Attached are output from 'netstat -an | find "LISTEN"' commands both before and after starting the server fresh with the different ports. I also ran with -o, but GIC Server shows as the system process so it's hard to distinguish anything extra with the process IDs.
[netstat_listeners_before_8091.txt](https://github.com/Terence-D/GameInputCommandServer/files/3929567/netstat_listeners_before_8091.txt)
[netstat_listeners_after_8091.txt](https://github.com/Terence-D/GameInputCommandServer/files/3929568/netstat_listeners_after_8091.txt)
[netstat_listeners_before_666.txt](https://github.com/Terence-D/GameInputCommandServer/files/3929569/netstat_listeners_before_666.txt)
[netstat_listeners_after_666.txt](https://github.com/Terence-D/GameInputCommandServer/files/3929570/netstat_listeners_after_666.txt)
[netstat_listeners_before_12366.txt](https://github.com/Terence-D/GameInputCommandServer/files/3929571/netstat_listeners_before_12366.txt)
[netstat_listeners_after_12366.txt](https://github.com/Terence-D/GameInputCommandServer/files/3929572/netstat_listeners_after_12366.txt)

**Comment by Terence-D** [comment]
Ok that sounds really weird.  It's been awhile since I've worked on this - busy rewriting the client - so it sounds like I'll need to take a deeper look and check my own install vs the logs you provided.  I'll leave this open until I can find out whats what.

Thanks for all the work you've done digging into this!

**Comment by SechyMonkey** [comment]
Happy to help. Let me know if you end up needing anything else from my end.

**Comment by Terence-D** [comment]
Leaving open due to the excellent logs, but migrating project to https://github.com/Terence-D/GamingInterfaceCommandServer
